src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/BlobReader.cs

```diff
@@ -591,9 +588,7 @@ public SignatureTypeCode ReadSignatureTypeCode()
             int length;
             if (TryReadCompressedInteger(out length))
             {
-                // Removal of trailing '\0' is a departure from the spec, but required
-                // for compatibility with legacy compilers.
-                return ReadUTF8(length).TrimEnd(s_nullCharArray);
```

Comment:
Were there any tests for this that need to be deleted?  Or any we want to add to ensure we're not trimming these?

Reply:
We'll see in the CI and I'll delete as needed.

I thought about adding a test, but do we also want to test that we don't trim after `\0` in the middle of the string or at the beginning then? (I'm not sure a test for this has a leg to stand on if we forget about the history of this - we don't generally test "absence of logic in the product that breaks stuff".

Reply:
> I'm not sure a test for this has a leg to stand on if we forget about the history of this

A comment can ensure we don't forget about the history :)

But I don't feel strongly about it.  I'm generally just questioning of any change that impacts product behavior and has zero test impact.